### PR TITLE
Add shardingsphere-all module to provide all-in-one jar

### LINKED ISSUE #13609: Consider about provide all-in-one jar
There are lots of released jars, could we consider about use a new release profile to provide all-in-one jar?

### PR DESCRIPTION
**Pull request by totalo**
Fixes #13609.

Changes proposed in this pull request:
- add shardingsphere-all module

### CODE REVIEW

**Comment by totalo** [comment]
image

**Review by terrymanu — CHANGES_REQUESTED** [review]
How about move shardingsphere-all into shardingsphere-distributed module?
And it is better to use maven profile to control assembly all-in-one jar or not.

**Comment by TeslaCN** [comment]
How to process the SPI files when shading?

**Comment by TeslaCN** [comment]
Netty is no longer re-package all classes into one jar. https://github.com/netty/netty/pull/11732

**Comment by totalo** [comment]
> How about move shardingsphere-all into shardingsphere-distributed module? And it is better to use maven profile to control assembly all-in-one jar or not.

All right, i will check it.

**Comment by totalo** [comment]
> How to process the SPI files when shading?

Merge them using ServicesResourceTransformer.

**Comment by totalo** [comment]
> Netty is no longer re-package all classes into one jar. [netty/netty#11732](https://github.com/netty/netty/pull/11732)

I think we can provide an uber.jar, as for other optional modules included in the user's classpath, the user needs to deal with it.

File: shardingsphere-distribution/pom.xml [hunk 1]
```diff
@@ -32,6 +32,7 @@
         <module>shardingsphere-src-distribution</module>
         <module>shardingsphere-jdbc-distribution</module>
         <module>shardingsphere-proxy-distribution</module>
+        <module>shardingsphere-all</module>
```
Comment (terrymanu) [inline shardingsphere-distribution/pom.xml:35]: It is better to rename as `shardingsphere-all-distribution`

**Comment by TeslaCN** [comment]
Hi @totalo @terrymanu 
Any update here? Could we finish this in 5.1.2?

**Comment by totalo** [comment]
> Hi @totalo @terrymanu Any update here? Could we finish this in 5.1.2?

Mybe not, put it to 5.1.3

**Comment by strongduanmu** [comment]
Hi @totalo, does this pr has any progress? Could we finish this in 5.1.3?

**Pull request closed without merging**